### ISSUE TITLE
Quest: Counterattack! Centaurs do not spawn

### ISSUE THREAD

**Issue by Virtualporygon2** [chromiecraft/chromiecraft#3439]
### What client do you play on?

enUS

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [X] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

When you start the quest the NPC Rethgar Deathgate will say ''Look to the west''. but no NPC's are spawning, Which means the mob carrying the Quest Objective is also not spawning because he only spawns when enough Kolkar Stormers and Kolkar Invaders have been killed.

### Expected Blizzlike Behaviour

Once you turn in H [5-30] [Hezrul Bloodmark](https://wowpedia.fandom.com/wiki/Hezrul_Bloodmark_(quest)), Regthar Deathgate will warn of an impending counter-attack by an army of Kolkar, which will immediately begin as soon as you accept the quest to end the life of the fourth and final Kolkar leader.

So long as you have the quest unfinished, a never-ending battle between Horde military and Kolkar warriors will ensue. The Kolkar troops will include [Kolkar Invaders](https://wowpedia.fandom.com/wiki/Kolkar_Invader) and [Kolkar Stormseers](https://wowpedia.fandom.com/wiki/Kolkar_Stormseer). By killing enough Kolkar attackers, the warlord will emerge. However, if you leave the invasion going for a certain amount of time, it will stop and you must re-activate it by speaking to Regthar again and selecting the option asking where Warlord Krom'zar is:

Gossip Where is Warlord Krom'zar?
I think I hear him coming! Look to the west!
Beware, <name>! Look to the west!
Whenever a Horde soldier is killed, Regthar will yell:

A defender has fallen!
Be warned, Krom'zar is a level 20 [elite mob](https://wowpedia.fandom.com/wiki/Elite_mob). Furthermore, he is guarded by his own group of Kolkar Invaders who will aggro alongside him no matter what. You shouldn't do this alone, despite the presence of Horde soldiers. Bring friends (or a willing high-level helper). Once Krom'zar is killed, [Krom'zar's Banner](https://wowpedia.fandom.com/wiki/Krom%27zar%27s_Banner) will appear atop his corpse and become lootable for  [[Piece of Krom'zar's Banner]](https://wowpedia.fandom.com/wiki/Piece_of_Krom%27zar%27s_Banner). The invasion will then cease, with Regthar declaring:

The Kolkar invaders are retreating!

### Source

https://wotlk.evowow.com/?quest=4021

https://wotlk.evowow.com/?item=11227

https://www.youtube.com/watch?v=5k6760g_M9E

### Steps to reproduce the problem

1. Accept the Quest: Counterattack! from Regthar Deathgate in the Barrens.
2. Kolkar invaders, Kolkar Stormers and Horde Soldiers should spawn, but nothing happens.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/59e45c251eb53abb0ed4b37a57254cbdf415e69a

### Operating system

Windows 11

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Server

ChromieCraft

**Comment by Dejanos** [comment]
Hello, Virtualporygon2. Upon first starting the quest on PTR I managed to do it with no problems and almost exactly like in [this Classic youtube video](https://youtu.be/5k6760g_M9E) but after giving it a second try, all I got is the opening dialogue like you with no Centaurs spawning, meaning the current behavior probably is that Elite needs to respawn in realm time or Horde defenders to despawn first in order for the quest to be able to be completed.

First time doing the quest:
![WoWScrnShot_043022_132527](https://user-images.githubusercontent.com/100415626/166104081-a40db925-2379-487d-b4bd-88f420f4d9cf.jpg)

![WoWScrnShot_043022_132536](https://user-images.githubusercontent.com/100415626/166104084-dbe9bcc0-8e23-45be-bcc5-efb5b8e9bcbf.jpg)

The second time after waiting for around 20 minutes after the first time no Centaurs spawned on start:

![WoWScrnShot_043022_133945](https://user-images.githubusercontent.com/100415626/166104105-9857f4ff-fc15-47a0-9ef6-019e0a740aa8.jpg)

In conclusion: Someone probably did the quest before you and quest initiation could not be triggered for some time after that, making you unable to complete your quest, which in my opinion should not be Blizzlike as even in the video I linked here [Horde Defenders despawned immediately after killing Warlord Krom'zar](https://youtu.be/5k6760g_M9E?t=351) making it available for the next group to initiate the quest.

**Comment by Dejanos** [comment]
Update: After more than 30 minutes and Horde Defenders despawning, Centaurs are not respawning in order for the quest to be done.
![WoWScrnShot_043022_140259](https://user-images.githubusercontent.com/100415626/166104762-8cc621e5-b339-45da-bdce-29c82f1f8396.jpg)

**Comment by Dejanos** [comment]
Update: After 3 hours and trying to start the quest again, it worked, it seems it just takes great intervals of time to pass for the quest to be doable again
![WoWScrnShot_043022_172343](https://user-images.githubusercontent.com/100415626/166112555-3420df92-5ab2-4d8e-bc3a-0a89a6eb7243.jpg)
.

**Comment by cachocarne** [comment]
This behavior persists as of October 7th, 2024. I managed to get centaurs to spawn after server restart or during low population times. I slowly killed centaurs until the Warlord appeared. However, quest cannot be completed, as there is a new problem: the banner is not spawning after killing the Warlord. He dies and that's it.

**Comment by stafoo** [comment]
Hello
I managed to complete this quest today, after 3-4 attempts.
Other times npcs were not spawning at all and battle never began.
The last time, npcs spawned and I was able to kill centaurs, then Warlord appeared and, after kill, the banner spawned and was clickable.

As a side note, only 2 or 3 centaurs were effectively fighting against orcs, instead the  other were just walking around doing nothing

**Comment by txtsd** [comment]
This bug still exists. The conversation option that is supposed to start spawning the centaurs does nothing at all so there's no way to proceed with or complete the quest.
